class_diagrams/core_architecture.md

```diff
@@ -0,0 +1,104 @@
+![](http://www.plantuml.com/plantuml/png/bLF1ZjCm4BtxAwm8YQwWAxX68TgbYm09QIMsWWiNhprfhRfsvPcqL8X_Pn8toWAXfhbOTgVllNblvWP9HMgtJY2Gz6eBk5CQ7c2deg8GPLaAt5ZVNRSIryg4W_KhsIY7yEGNxHIY_24aVs-LtgnYQBrP11TYLKKmhOPOr_93y72yWyDqdT9aW-UohsjGD5s3u35qK4lGQyzlg-DquTx2WSlkervRCrq2JGEQkEXzSEPVT8vBUi9FQW_UG1G_XNZ-PloaWSPwsqb7OJr1RDX1MIn217dRqHD3j17AKjxs7LRBkf1ZkvrjB-YuWn8fihbyCufKkMIc7E3dP6qXHvSJNhhDa1H1AH-wuwLCZ8ECgzDxbeRhUjUPPV0_aGJ_aHkVNPNxO4r_lmFk8Ho7fppokpu3HXxJOEHvKehFolj7XBXNZmwgf1hh_zAROcFSCSAa6Rw3RWbO8GxDP-6-y2mnL9gfmTzk5fxqcJPWMWspG_V_XbS53Y5kJY1LjOYW2FZQrynE6MBN0LMdSEihDgGN-viTVNPrpYkU9SihJNez-oFz8RJapsomQJJlv52ckKNm58Dp42Sw_6jNdFKR5EASJSLh1fhF-Ro-zaLcx4jPIwsSGt5sRejNZ3N05cTIIeY3PP23pKiTLpv1jyYWPpVXGgtE8e57IK4gR-GtFuhdDv7DZ3akvBeCDtcJMTECDPVXfdrDc6bDKTpmsMxTBm00)
+
+# Core Architecture
+
+The core architecture is based on Event Sourcing (Young, 2010) and Command and Query Responsibility Segregation (CQRS) (Meyer, 1994) and a modified Model-View-Controller (MVC) (Reenskaug, 1979). 
```

Comment:
I'm missing the bibliography entry for Greg Young. I'm trying to find the original presentation he gave but all I can currently find is second-hand accounts. I also need to run this through Grammarly. 😬

Reply:
Need to add a bit in about the state (and the decision between using the reducer to handle data manipulation or the selector).